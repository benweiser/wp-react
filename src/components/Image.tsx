import * as React from "react";
import LazyLoad from "react-lazyload";
import Img from "react-image";
import ReactLoading from "react-loading";
import styled from "styled-components";

export interface ImageProps {
  alt: string;
  src: string;
  className?: string;
}

const Image: React.SFC<ImageProps> = ({
  className,
  alt,
  src
}: ImageProps): JSX.Element => {
  return (
    <StyledImage>
      <LazyLoad offset={100} height={200} once={true}>
        <Img
          className={className}
          src={[src, "http://via.placeholder.com/350x350"]}
          alt={alt}
          loader={<ReactLoading type={"bubbles"} color={"#ccc"} />}
        />
      </LazyLoad>
    </StyledImage>
  );
};

const StyledImage = styled.div`
/*  padding-bottom: 100%;
  position: relative;
  height: 0;
  > img {
    position: absolute;
    top: 0;
    left: 0;*/
  }
`;

export default Image;
