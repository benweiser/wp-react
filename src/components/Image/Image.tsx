import * as React from "react";
import LazyLoad from "react-lazyload";
import Img from "react-image";
import { Loader } from "../atoms/Loader/Loader";

export interface ImageProps {
  readonly alt: string;
  readonly src: string;
  readonly className?: string;
}

const Image: React.SFC<ImageProps> = ({
  className,
  alt,
  src
}: ImageProps): JSX.Element => {
  return (
    <div>
      <LazyLoad offset={100} height={200} once={true}>
        <Img
          className={className}
          src={[src, "http://via.placeholder.com/350x350"]}
          alt={alt}
          loader={<Loader />}
        />
      </LazyLoad>
    </div>
  );
};

export default Image;
