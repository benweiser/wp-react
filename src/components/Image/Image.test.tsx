import { shallow } from "enzyme";
import Image from "./Image";
import * as React from "react";
import toJson from "enzyme-to-json";

const logo = require("../../assets/images/logo.png");

it("Image component renders correctly", () => {
  const wrapper = shallow(
    <Image src={logo} alt={"al text"} className={"bw-img"} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
