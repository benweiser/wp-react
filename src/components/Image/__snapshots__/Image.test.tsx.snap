// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component renders correctly 1`] = `
<div>
  <LazyLoad
    height={200}
    offset={100}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <Img
      alt="al text"
      className="bw-img"
      decode={true}
      loader={<Unknown />}
      src={
        Array [
          "logo.png",
          "http://via.placeholder.com/350x350",
        ]
      }
      unloader={false}
    />
  </LazyLoad>
</div>
`;
